src/test/scala/fr/davit/capturl/IriSpec.scala

```diff
@@ -117,4 +119,20 @@ class IriSpec extends FlatSpec with Matchers {
 
   }
 
+  "LazyIri" should "parse at least everything that StrictIri parses" in {
+    val url = "http://example.com/path?query#fragment"
+    Iri(url, Lazy) shouldBe Iri(url, Strict)
+  }
```

Comment:
Strictly speaking, this is not what you are testing. What's the exact value of this test?

Reply:
To have a base case where the lazy and the strict version show identical behaviour, but I agree that this test probably doesn't bring a lot of value.

Reply:
I deleted the test case